domain/src/main/java/org/oppia/android/domain/platformparameter/PlatformParameterController.kt

```diff
@@ -0,0 +1,53 @@
+package org.oppia.android.domain.platformparameter
+
+import org.oppia.android.app.model.PlatformParameter
+import org.oppia.android.app.model.RemotePlatformParameterDatabase
+import org.oppia.android.data.persistence.PersistentCacheStore
+import org.oppia.android.domain.oppialogger.OppiaLogger
+import org.oppia.android.util.data.DataProvider
+import org.oppia.android.util.data.DataProviders.Companion.transform
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val PLATFORM_PARAMETER_DATA_PROVIDER_ID = "platform_parameter_data_provider_id"
+
+@Singleton
+class PlatformParameterController @Inject constructor(
+  cacheStoreFactory: PersistentCacheStore.Factory,
+  private val oppiaLogger: OppiaLogger,
```

Comment:
I think we want to use ConsoleLogger here, I believe OppiaLogger is meant for app events while ConsoleLogger is for Android logcat & file logging.

Reply:
You can use OppiaLogger for console logging too..I think stuff like oppiaLogger.d should log to console so we are fine here!

Reply:
Yeah. Also coincidently I was the one who have changed the flow of using OppiaLogger instead of ConsoleLogger in this PR  - https://github.com/oppia/oppia-android/pull/3104

Reply:
Haha oops, how funny